### ISSUE TITLE
Add Android helper to upload a Bitmap to a texture

### ISSUE THREAD

**Issue by romainguy** [google/filament#314]
For the Java/Kotlin APIs, to avoid a copy to a `ByteBuffer`.

**Comment by romainguy** [comment]
This feature has been implemented.

**Comment by rawnsley** [comment]
I'm seeing a problem when calling _TextureHelper.setBitmap_ and then immediately disposing of the Bitmaps on the JVM side (calling _Bitmap.recycle_ and _System.gc_). The textures tend to render incorrectly or I get an app crash.

I can't see anything to protect against this in the Filament code, but I'm also surprised that it isn't protected by default in [_AndroidBitmap_lockPixels_](https://developer.android.com/ndk/reference/group/bitmap#androidbitmap_lockpixels). Perhaps I'm missing something subtle?

It happens every time in my app, but I couldn't make it happen in the textured-object sample, possibly because the single bitmap is not overwritten by subsequent bitmaps.

Even if we don't explicitly recycle or garbage collection, there's a risk the JVM will do it anyway.

**Comment by romainguy** [comment]
You cannot dispose of the texture right away. The upload is asynchronous because of the render thread. We offer a callback mechanism when uploading a texture to be notified of when the bitmap can be safely discarded. The texture upload code keeps a strong reference though, so as long as you don't call dispose() yourself (which I don't recommend doing anyway), you'll be fine.

**Comment by rawnsley** [comment]
That makes sense - thanks for the info.

**Comment by rawnsley** [comment]
@romainguy Sorry to wake up an old thread, but I can't seem to get this to work. If I use [setImage](https://github.com/google/filament/blob/master/android/filament-android/src/main/java/com/google/android/filament/Texture.java#L424) and pass a _PixelBufferDescriptor_ with a callback, the callback never fires. Is there an example of how this might be done from the JVM side?

Also the callback mechanism doesn't appear to be available when using _TextureHelper.setBitmap_, which is not a huge problem because I don't have to use it.

I know I should leave disposal to the garbage collector, but on older devices it sometimes doesn't get around to it in time to prevent OOM and also Kotlin has a nasty habit of keeping objects alive if they are returned from coroutines.

**Comment by romainguy** [comment]
I just tested using a callback with a `PixelBufferDescriptor` and it worked just fine. I modified one of the samples to print a message when the IBL is properly loaded:

```
val buffer = Texture.PixelBufferDescriptor(storage,
            Texture.Format.RGB, Texture.Type.UINT_10F_11F_11F_REV, 1, 0, 0, 0,
            Handler(Looper.getMainLooper())) {
        android.util.Log.d("Filament", "Cubemap transferred")
}
```
I can see the message appear twice for the two cubemaps loaded by the sample.

A callback would not be very useful for `setBitmap()`. Trying to manually discard a bitmap works differently across versions of Android and one some it just doesn't do anything (or at least anything that helps).
